app/src/main/java/com/example/candyshop/MainActivity.kt

```diff
@@ -3,44 +3,168 @@ package com.example.candyshop
+    Row(
         modifier = modifier
+            .padding(4.dp)
+            .fillMaxWidth()
+            .border(2.dp, Color(0xFFD3CEDF))
+            .size(width = 400.dp, height = 120.dp)
+            .padding(4.dp),
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        Image(
+            painter = painterResource(id = imgId),
+            contentDescription = null,
+            contentScale = ContentScale.Crop,
+            modifier = Modifier
+                .size(120.dp)
+                .border(2.dp, Color(0xFF9CB4CC), CircleShape)
+                .padding(2.dp)
+                .clip(CircleShape)
+        )
+        Column(
+            modifier = Modifier.padding(start = 16.dp)
+        ) {
+            Text(
+                text = name,
+                fontSize = 20.sp,
+                fontWeight = FontWeight.Bold,
+                color = Color(0xFF748DA6)
+            )
+            Text(
+                text = "price: $price",
+                color = Color(0xFF748DA6),
+                modifier = Modifier.padding(top = 6.dp, bottom = 14.dp)
+            )
+            Button(
+                onClick = {
+                    navController.navigate(Screen.DetailsScreen.withArgs(name))
+                },
+            colors = ButtonDefaults.buttonColors(
+                containerColor = Color(0xFFD3CEDF),
+                contentColor = Color(0xFF748DA6)),
+            border = BorderStroke(2.dp, Color(0xFF748DA6)),
+            modifier = Modifier
+                .widthIn(min = 250.dp)
+                .heightIn(max = 35.dp)
+            ) {
+                Text("DETAILS")
+            }
+        }
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun TopLogoBar(modifier: Modifier = Modifier) {
+    CenterAlignedTopAppBar(
+        title = {
+            Row(
+                verticalAlignment = Alignment.CenterVertically,
+                modifier = modifier
+            ) {
+                Icon(
+                    painter = painterResource(id = R.drawable.icecream_icon),
+                    contentDescription = "ice cream icon",
+                    tint = Color(0xFF748DA6)
+                )
+                Spacer(Modifier.size(ButtonDefaults.IconSpacing))
+                Text(
+                    text = "CANDY SHOP",
+                    fontSize = 24.sp,
+                    fontWeight = FontWeight.Bold,
+                    color = Color(0xFF748DA6)
+                )
+                Spacer(Modifier.size(ButtonDefaults.IconSpacing))
+                Icon(
+                    painter = painterResource(id = R.drawable.icecream_icon),
+                    contentDescription = "ice cream icon",
+                    tint = Color(0xFF748DA6)
+                )
+            }
+        }
     )
 }
 
-@Preview(showBackground = true)
 @Composable
-fun GreetingPreview() {
-    CandyShopTheme {
-        Greeting("Android")
+fun CandyShopMain(navController: NavController) {
```

Comment:
we could pass this list directly to CandyShopMain changing it to:
fun CandyShopMain(
    navController: NavController,
    candies: List<ContentItem>)

If we change source of data to dynamic one we will have to anyway do it it this way, it also allows you to do more variations for previews